src/main/java/com/arpnetworking/utility/RegexAndMapReplacer.java

```diff
@@ -0,0 +1,296 @@
+/**
+ * Copyright 2018 Inscope Metrics, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.utility;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A regex replacement utility that can also replace tokens not found in the regex.
+ *
+ * $n where n is a number in the replace string is replaced by the pattern's match group n
+ * ${name} in the replace string is replaced by the pattern's named capture, or by the value of the variable
```

Comment:
This documentation doesn't match the documentation in the README. Also, I thought the numeric support was going to be removed altogether. Having the $0 format supported begs the question why not support $foo as well with white-space parsing a la bash. I think it's a path we shouldn't go down.